pkg/lwip/include/sock_types.h

```diff
@@ -127,6 +130,9 @@ struct sock_tcp_queue {
  */
 struct sock_udp {
     lwip_sock_base_t base;          /**< parent class */
+    mutex_t mutex;                  /**< Mutex to protect the sock */
+    struct netbuf *last_buf;        /**< Last received data */
+    bool peek_buf_avail;            /**< Whether there is a buffer left from a previous peek operation */
```

Comment:
same here

Reply:
btw is there a connection between `last_buf` and `peek_buf_avail`?

Reply:
We keep `last_buf` always in the sock as is until it is received without the peek flag. `peek_buf_avail` is basically used to keep track of whether we already read from `last_buf` or not which is important for `sock_ip_recv_aux` which is looping over `sock_ip_recv_buf_aux`. The first call of `sock_ip_recv_buf_aux` will return the peeked data and a result `> 0`. When calling `sock_ip_recv_buf_aux` a second time (since the result was `> 0`) we reset the flag and return `0` to break the loop. Returning `0` instantly would be another option, but that does not seem right.